mod/workspace/cache.js

```diff
@@ -81,38 +81,50 @@ async function cache() {
     JSON.stringify(workspace.locales).replace(
       /\$\{(.*?)\}/g,
       (matched) =>
-        process.env[`SRC_${matched.replace(/\$|\{|\}/g, '')}`] || matched
+        process.env[`SRC_${matched.replace(/\$|\{|\}/g, "")}`] || matched
     )
   );
 }
 
 async function getDynamicWorkspace(req) {
-  if (!req) return new Error('Not request found');
-  const { pageId, project, lang } = req.query;
-  const { mongoClient } = req;
-  if (!pageId) {
-    return new Error('No pageId specified');
-  }
-  if (!project) {
-    return new Error('No project specified');
-  }
-  if (!lang) {
-    return new Error('No lang specified');
+  try {
+    if (!req) return new Error("Not request found");
+    const { pageId, project, lang } = req.query;
+    const { mongoClient } = req;
+    if (!pageId) {
+      return new Error("No pageId specified");
+    }
+    if (!project) {
+      return new Error("No project specified");
+    }
+    if (!lang) {
+      return new Error("No lang specified");
+    }
+    const { ObjectId } = require("mongodb");
+    const Page = mongoClient.db("acorn").collection("pages");
+    const Version = mongoClient.db("acorn").collection("versions");
+    const proposalPage = await Page.findOne({ _id: ObjectId(pageId) });
```

Comment:
Added the `ObjectId()` into `pageId` const

Reply:
Nice catch!